include/userobjects/BooleanComboClusteringUserObject.h

```diff
@@ -0,0 +1,76 @@
+#pragma once
+
+#include "GeneralUserObject.h"
+#include <unordered_map>
+
+class ClusteringHeuristicUserObjectBase;
+
+/**
+ * A clustering user object which implements the mesh walking process
+ * and implements the clustering by evaluating  a user defined
+ * boolean expression involving other heuristic based user objects
+ */
+class BooleanComboClusteringUserObject : public GeneralUserObject
+{
+public:
+  static InputParameters validParams();
+  BooleanComboClusteringUserObject(const InputParameters & parameters);
+
+  virtual void execute() override;
+  virtual void initialize() override {};
+  virtual void finalize() override {};
+
+private:
+  /**
+   * getter function for extra element integer score of an element
+   * @param[int] a libmesh element
+   * @param[out] extra element integer score of that element
+   */
+  int getExtraIntegerScore(libMesh::Elem * elem) const;
+
+  /**
+   * function for converting the expression input to reverse polish notation
+   * it follows the  shunting yard algorithm.
+   * @param[in] a parsed expression
+   */
+  void reversePolishNotation(const std::vector<std::string> & expression);
+
+  /**
+   * Method for evaluating if element should be clustered or not
+   * @param[in] base_element, a neighboring element of the base element
```

Comment:
Need to have one line per input argument, they cannot be grouped like this I do not think it works for Doxygen.

Please check other Doxygen comments for consistency.